sriovnet.go

```diff
@@ -520,3 +520,37 @@ func GetPciFromNetDevice(name string) (string, error) {
 	}
 	return base, nil
 }
+
+// GetPKeyByIndexFromPci returns the PKey stored under given index for the IB PCI device
+func GetPKeyByIndexFromPci(pciAddress string, index int) (string, error) {
+	pciDir := filepath.Join(PciSysDir, pciAddress, "infiniband")
```

Comment:
question:

should we just try to read: `/sys/bus/pci/devices/<pciAddr>/infiniband/ports/1/pkeys/<index>` file straight away ?
just to save a bunch of filesystem operations.

also im not aware of devices that have multiple infiniband ports under the same PCI function.
(there was in the distant past but no more...)

Reply:
We actually need the name of the RDMA device to construct the path. However, I've updated the func to always use the port number 1